genesyscloud/resource_genesyscloud_credential_test.go

```diff
@@ -0,0 +1,161 @@
+package genesyscloud
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+	"testing"
+
+	"github.com/google/uuid"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+	"github.com/mypurecloud/platform-client-sdk-go/platformclientv2"
+)
+
+func TestAccResourceCredential(t *testing.T) {
+	var (
+		credResource1 = "test_credential_1"
+		credResource2 = "test_credential_2"
+		credName1     = "Terraform Credential Test-" + uuid.NewString()
+		credName2     = "Terraform Credential Test-" + uuid.NewString()
+
+		typeName1 = "basicAuth"
+		typeName2 = "callJourney"
+
+		key1 = "userName"
+		val1 = "someUserName"
+
+		key2 = "authToken"
+		val2 = "fakeToken"
+	)
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:          func() { testAccPreCheck(t) },
+		ProviderFactories: providerFactories,
+		Steps: []resource.TestStep{
+			{
+				// Create
+				Config: generateCredentialResource(
+					credResource1,
+					strconv.Quote(credName1),
+					strconv.Quote(typeName1),
+					generateCredentialFields(
+						generateMapKeyValue(key1, strconv.Quote(val1)),
+					),
+				),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("genesyscloud_credential."+credResource1, "name", credName1),
+					resource.TestCheckResourceAttr("genesyscloud_credential."+credResource1, "credential_type_name", typeName1),
+					resource.TestCheckNoResourceAttr("genesyscloud_credential."+credResource1, "fields"),
+				),
+			},
+			{
+				// Update name
+				Config: generateCredentialResource(
+					credResource1,
+					strconv.Quote(credName2),
+					strconv.Quote(typeName1),
+					generateCredentialFields(
+						generateMapKeyValue(key1, strconv.Quote(val1)),
```

Comment:
Can we test updating the fields? That would be the most likely change I'm guessing.

Reply:
Yeah I can add some key-value pairs and also change values for the same keyname. But I guess I can change them only to check if the program runs? Since it cannot be set by `d.Set()`, there is no way to actually test it